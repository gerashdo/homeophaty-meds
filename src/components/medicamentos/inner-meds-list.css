
.delete-btn {
    display: none;
}

li:hover .delete-btn {
    display: block;
    transition: all 0.3s ease-in-out;
}