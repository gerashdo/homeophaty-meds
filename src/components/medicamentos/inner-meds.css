


input.inner-input {
    padding: 2px 5px;
    border: 0;
    border-bottom: 1px solid rgb(169, 169, 169);
    width: 100%;
    outline: none;
}

.input.process button {
    border: none;
    background-color: var(--color-primary);
    color: white;
    padding: 0.125rem 0.5rem;
    border-radius: 25%;
    cursor: pointer;
}