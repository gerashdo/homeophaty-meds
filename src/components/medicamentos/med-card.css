.med-card{
    /* border: 1px solid rgb(0, 0, 0); */
    border-radius: 1.25rem;
    box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.1);
    display: flex;
    flex-direction: column;
    height: 20rem;
    gap: 1rem;
    justify-content: space-between;
    min-width: 60%;
    transition: all 0.1s ease;
    overflow: hidden;
}

.med-card:hover {
    box-shadow: 2px 1px 5px rgba(0, 0, 0, 0.2);
    min-width: 62%;
}

.med-card-content {
    display: flex;
    flex-direction: column;
    gap: 1rem;
}


.med-card .med-card-content .title {
    font-weight: var(--font-bold);
    font-size: var(--font-size-subtitle);
}

.med-card .title{
    color: white;
    padding: 1rem;
}

.med-card .title.formula {
    background-color: var(--color-secondary-darker);
}

.med-card .title.normal {
    background-color: var(--color-primary-darker);
}

.med-card-content p{
    color: var(--color-black-lighter);
    padding: 0 2rem;
    text-align: start;

}

.med-card .card-footer{
    padding: 1rem;
}

button a {
    text-decoration: none;
    color: white;
}