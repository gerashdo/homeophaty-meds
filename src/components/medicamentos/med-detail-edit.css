
div.med-data{
    display: flex;
    align-items: flex-start;
    gap: 3rem;
    justify-content: center;
}

.med-titlename input{
    background-color: var(--color-gray-background);
    border: 0;
    border-radius: 1rem;
    font-size: var(--font-size-subtitle);
    font-weight: var(--font-bold);
    margin: 1rem 0;
    padding: 0.5rem 1rem;
    text-align: center;
}

.med-titlename input:focus{
    outline: none;
}

div.med-data h4{
    text-align: start;
    margin: 1rem 0 1rem 0.25rem;
    font-weight: var(--font-bold);
}

div.med-data div.inner-med-list {
    min-width: 30%;
}

div.med-data ul {
    width: 100%;
}

div.med-data form {
    min-width: 50%;
    width: 80%;
}

div.med-data form textarea {
    width: 100%;
    border: none;
    border-radius: 0.5rem;
    background-color: var( --color-gray-background);
    resize: none;
    padding: 1rem;
}

.submit-button-container{
    display: flex;
    justify-content: flex-end;
    margin-top: 1rem;
}

@media (max-width: 700px){
    div.med-data{
        display: flex;
        flex-direction: column;
        gap: 2rem;
        width: 100%;
    }

    div.inner-med-list {
        width: 100%;
    }

    .med-titlename h3 {
        font-size: var(--font-size-subtitle);
        font-weight: var(--font-bold);
        margin: 1rem 0;
    }

    div.med-data form {
        width: 100%;
    }
}