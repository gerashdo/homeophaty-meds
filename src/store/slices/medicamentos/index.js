export * from './medicamentoSlice'
export * from './thunks'