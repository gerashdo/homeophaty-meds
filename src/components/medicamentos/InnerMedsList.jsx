import React from 'react'
import { CgTrash } from 'react-icons/cg'

import './inner-meds-list.css'

export const InnerMedsList = ({ innerMeds, onRemoveInnerMeds }) => {
  return (
    <ul className='mb-4 flex gap-3'>
      {innerMeds.map(med => (
        <li key={med._id} className='flex border-2 border-gray-300 px-3 py-1 rounded-md  items-center justify-between'>
          <span>{med.name} {med.ch}</span>
          <button className='delete-btn text-gray-400 hover:text-gray-600 p-1' onClick={() => onRemoveInnerMeds(med._id)}><CgTrash /></button>
        </li>
      ))}
    </ul>
  )
}
