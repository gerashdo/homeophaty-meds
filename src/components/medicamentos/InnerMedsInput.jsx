import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types'
import { useForm } from '../../hooks/useForm';
import { startAlert } from '../../store/slices/ui';
import { DinamicSelector } from '../iterface/DinamicSelector';
import { searchStringInMed } from '../../helpers';

import './inner-meds.css'

// Parametros
// onChangeInnerMeds: Accion a realizar cuando se agregue un nuevo medicamento
// innerMeds: Lista de innerMeds actuales
export const InnerMedsInput = ({ onChangeInnerMeds, innerMeds = [] }) => {
    const dispatch = useDispatch()
    const { medicamentos } = useSelector( state => state.medicamento )
    const [ values, handleChange, setValues, reset ] = useForm({ medicamento: '' })
    const { medicamento } = values
    const [ focused, setFocused ] = useState( false )

    const changeInputValue = ( newValue ) => {
        setValues({
            ...values,
            medicamento: newValue
        })
    }

    const handleAddMedicine = (e) => {
        e.preventDefault()

        // Si el medicamento ya esta dentro del los inner meds
        if( searchStringInMed( medicamento, innerMeds ).length > 0 ){
            return dispatch( startAlert({
                alertMessage: 'El medicamento ya se encuentra en la lista',
                alertType: 'info'
            }) )
        }

        const medsFound = searchStringInMed( medicamento, medicamentos )

        // Si hay mas de un resultado con el texto introducido en el input
        if( medsFound.length > 1 ){
            return dispatch( startAlert({
                alertMessage: 'Hay m√°s de un medicamento con el nombre que introdujo',
                alertType: 'info'
            }) )
        }
        
        onChangeInnerMeds( medsFound[0] )

        reset()
    }

    return (
        <div 
            className='horizontal-input-button'
        >
        <div className='input process relative'>
            <input
                name='medicamento'
                className='inner-input'
                onChange={ handleChange }
                value={ medicamento }
                onFocus={ () => setFocused( true ) }
                onBlur={ () => setFocused( false ) }
            />
            {
                focused
                &&
                <DinamicSelector 
                    resetValues={ medicamentos }
                    valueForFilter={ medicamento }
                    onSetValue={ changeInputValue }
                />
            }
        </div>
            <button
                className='primary'
                onClick={ handleAddMedicine }
            >
                +
            </button>
        </div>
    )
}

InnerMedsInput.propTypes = {
    onChangeInnerMeds: PropTypes.func.isRequired,
    innerMeds: PropTypes.arrayOf( PropTypes.object )
}