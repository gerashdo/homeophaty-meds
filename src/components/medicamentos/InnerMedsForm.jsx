import { InnerMedsInput } from "./InnerMedsInput"
import { MedSmallCard } from "./MedSmallCard"
import PropTypes from "prop-types"

export const InnerMedsForm = ({ innerMeds = [], onRemoveInnerMeds, onChangeInnerMeds, label = '' }) => {

    const handleOnChangeInnerMeds = ( medicine ) => {
        onChangeInnerMeds( medicine )
    }

    return (
        <>
            {
                innerMeds.length > 0 
                ? (
                    <div>
                        <ul>
                            {
                                innerMeds.map( med => (
                                    <MedSmallCard 
                                        key={ med.id } 
                                        medicamento={ med }
                                        onCloseInnerMed={ onRemoveInnerMeds }
                                        isDelete={ true }
                                    />
                                ))
                            }
                        </ul>
                    </div>
                )
                : null
            }

            <div>
                {
                    label.length > 0
                    ? ( <label htmlFor="inner_meds">{ label }</label> )
                    : null
                }
                <InnerMedsInput 
                    id="inner_meds"
                    onChangeInnerMeds={ handleOnChangeInnerMeds }
                    innerMeds={ innerMeds }
                />
            </div>
        </>

    )
}

InnerMedsForm.propTypes = {
    innerMeds: PropTypes.array,
    onRemoveInnerMeds: PropTypes.func.isRequired,
    onChangeInnerMeds: PropTypes.func.isRequired,
    label: PropTypes.string,
}