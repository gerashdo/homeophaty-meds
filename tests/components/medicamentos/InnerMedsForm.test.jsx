import { configureStore } from "@reduxjs/toolkit";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { InnerMedsForm } from "../../../src/components/medicamentos/InnerMedsForm";
import { medicamentoSlice } from "../../../src/store/slices/medicamentos";
import { abrotanum30, medsListFixture } from "../../fixtures/meds";

const getMockStore = ( initialState ) => {
    return configureStore({
        reducer: {
            medicamento: medicamentoSlice.reducer,
        },
        preloadedState: {
            medicamento: { ...initialState }
        }
    })
}

describe('Tests for <InnerMedsForm /> component', () => {
    const onRemoveInnerMeds = jest.fn()
    const onChangeInnerMeds = jest.fn()

    test('should render the component correctly', () => {
        const mockStore = getMockStore({ medicamentos: medsListFixture })

        const { container } = render( <InnerMedsForm 
            innerMeds={ medsListFixture }
            onRemoveInnerMeds={ onRemoveInnerMeds }
            onChangeInnerMeds={ onChangeInnerMeds }
        />, {
            wrapper: ({ children }) => <Provider store={ mockStore }>{ children }</Provider>
        })

        expect( container ).toMatchSnapshot()
    });

    test('should call onChangeInnerMeds if a medicine is added to the inner meds', () => {
        const mockStore = getMockStore({ medicamentos: medsListFixture })

        const { queryByRole } = render( <InnerMedsForm
            onRemoveInnerMeds={ onRemoveInnerMeds }
            onChangeInnerMeds={ onChangeInnerMeds }
        />, {
            wrapper: ({ children }) => <Provider store={ mockStore }>{ children }</Provider>
        })

        const input = queryByRole('textbox')
        act( () => {
            input.focus()
            fireEvent.change( input, { target: { value: `${ abrotanum30.name }` }} )
            queryByRole('button').click()
        })

        expect( onChangeInnerMeds ).toHaveBeenCalledWith( abrotanum30 )
    });
});