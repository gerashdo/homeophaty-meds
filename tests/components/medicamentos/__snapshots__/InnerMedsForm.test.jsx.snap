// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <InnerMedsForm /> component should render the component correctly 1`] = `
<div>
  <div>
    <ul>
      <div
        class="med-small-card"
      >
        <div>
          Abrotanum
           
          30
        </div>
        <div
          class="cancel-button"
          data-testid="cancel-button"
          id="1"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM5.354 4.646a.5.5 0 1 0-.708.708L7.293 8l-2.647 2.646a.5.5 0 0 0 .708.708L8 8.707l2.646 2.647a.5.5 0 0 0 .708-.708L8.707 8l2.647-2.646a.5.5 0 0 0-.708-.708L8 7.293 5.354 4.646z"
            />
          </svg>
        </div>
      </div>
      <div
        class="med-small-card"
      >
        <div>
          Calendula
           
          200
        </div>
        <div
          class="cancel-button"
          data-testid="cancel-button"
          id="3"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM5.354 4.646a.5.5 0 1 0-.708.708L7.293 8l-2.647 2.646a.5.5 0 0 0 .708.708L8 8.707l2.646 2.647a.5.5 0 0 0 .708-.708L8.707 8l2.647-2.646a.5.5 0 0 0-.708-.708L8 7.293 5.354 4.646z"
            />
          </svg>
        </div>
      </div>
      <div
        class="med-small-card"
      >
        <div>
          Calendula
           
          30
        </div>
        <div
          class="cancel-button"
          data-testid="cancel-button"
          id="2"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM5.354 4.646a.5.5 0 1 0-.708.708L7.293 8l-2.647 2.646a.5.5 0 0 0 .708.708L8 8.707l2.646 2.647a.5.5 0 0 0 .708-.708L8.707 8l2.647-2.646a.5.5 0 0 0-.708-.708L8 7.293 5.354 4.646z"
            />
          </svg>
        </div>
      </div>
    </ul>
  </div>
  <div>
    <div
      class="horizontal-input-button"
    >
      <div
        class="input process relative"
      >
        <input
          class="inner-input"
          name="medicamento"
          value=""
        />
      </div>
      <button
        class="primary"
      >
        +
      </button>
    </div>
  </div>
</div>
`;
